jina/optimizers/__init__.py

```diff
@@ -9,17 +9,16 @@
 
 class EvaluationCallback:
     def __init__(self, eval_name=None):
-        self.op_name = eval_name
+        self.eval_name = eval_name
         self.evaluation_values = {}
         self.n_docs = 0
 
-    def flush(self):
-        self.evaluation_values = {}
-        self.n_docs = 0
+    def get_fresh_callback(self):
```

Comment:
I would have this logic in the caller of this method and not as a method here.

Reply:
Isn't this a general pattern of callback class we would want the user and our developers to follow?